### PR TITLE
[DRAFT] forgejo: init at 1.18.0-rc1-1

### PR DESCRIPTION
**Pull request by Infinidoge**
###### Description of changes

[Forgejo](https://forgejo.org) is a new fork of [Gitea](https://gitea.io) by the people at [Codeberg](https://codeberg.org), designed to try and make a new place where the Gitea community can come together and control their project again. Codeberg themselves have stated that once Forgejo releases its first stable version, they will be switching over.

Forgejo is a soft-fork as of currently, so it shares the vast majority of its code base with Gitea, so most of the work done in this PR is just duplicating the existing things for Gitea and editing text. That being said, since Forgejo has not yet released its first stable release, I intend on keeping this PR as a draft until that happens. (Also allows for time for feedback, and for making finishing touches.)

Announcement Blog Posts:
https://blog.codeberg.org/codeberg-launches-forgejo.html
https://forgejo.org/2022-12-15-hello-forgejo/

###### TODO
- [ ] Duplicate Gitea tests
- [ ] Add release notes entry for module
- [ ] Bump to first stable release when it happens
- [ ] Change URL to not need to be manually replaced each update (See https://codeberg.org/forgejo/forgejo/issues/128)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/forgejo/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  goPackagePath = "code.gitea.io/gitea";
+
+  passthru.tests = nixosTests.gitea;
+
+  meta = with lib; {
+    description = "Beyond coding. We forge.";
+    homepage = "https://forgejo.org";
+    license = licenses.mit;
+    maintainers = with maintainers; [ disassembler kolaente ma27 techknowlogick ];
```
Comment (Ma27) [inline pkgs/applications/version-management/forgejo/default.nix:73]: Please don't add me as maintainer in new packages without asking.

Comment (kolaente) [reply pkgs/applications/version-management/forgejo/default.nix:73]: I second this.

Comment (Infinidoge) [reply pkgs/applications/version-management/forgejo/default.nix:73]: Ah, whoops, I changed the maintainers on the module but forgot to on the package.

**Comment by adamcstephens** [comment]
1.18.0-1 is now released. https://codeberg.org/forgejo/forgejo/releases/tag/v1.18.0-1

**Comment by adamcstephens** [comment]
Actually this package had already been merged in #207796

**Comment by trepetti** [comment]
> Actually this package had already been merged in #207796
That PR didn't have the service, however. Would it be possible to rebase this to after the base package got merged to keep the service functionality?

**Comment by adamcstephens** [comment]
I agree we should have a separate module for Forgejo as well, sorry I missed this had that functionality. For now you should be able to use `services.gitea.package = pkgs.forgeo`

**Comment by Infinidoge** [comment]
Went ahead and did the rebase and pulled out specifically the service functionality.
Was doing to rename and reopen this PR, but I can't actually reopen it, so I'm going to go ahead and make a new PR.

**Pull request closed without merging**